File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/capabilities/CapabilityFactory.java
package au.com.fullcirclesolutions.saucery.capabilities;

import org.openqa.selenium.remote.DesiredCapabilities;

public class CapabilityFactory {
    public static DesiredCapabilities CreateCapabilities(SaucePlatform platform, String testName) {
        String sanitisedTestName = Sanitiser.SanitiseTestName(testName, platform);
        if (platform.IsADesktopPlatform()) {
            return (new DesktopCreator()).Create(platform, sanitisedTestName).GetCaps();
        }
        //Mobile Platform
        return platform.CanUseAppium()
                //Mobile Platform
                ? platform.IsAnAppleDevice()
                        ? (new AppiumIOSCreator()).Create(platform, sanitisedTestName).GetCaps()
                        : (new AppiumAndroidCreator()).Create(platform, sanitisedTestName).GetCaps()
                //Devolve to WebDriver    
                : platform.IsAnAppleDevice()
                        ? (new WebDriverIOSCreator()).Create(platform, sanitisedTestName).GetCaps()
                        : (new WebDriverAndroidCreator()).Create(platform, sanitisedTestName).GetCaps();
    }
}
/*
 * Copyright: Andrew Gray, Full Circle Solutions
 * Date: 10th August 2014
 */

File: NUnit/Version2/Saucery2/Capabilities/CapabilityFactory.cs
using OpenQA.Selenium.Remote;
using Saucery2.Capabilities.ConcreteCreators;
using Saucery2.OnDemand;

namespace Saucery2.Capabilities {
    internal class CapabilityFactory {
        public static DesiredCapabilities CreateCapabilities(SaucePlatform platform, string testName) {
            if (platform.IsADesktopPlatform()) {
                return (new DesktopCreator()).Create(platform, testName).GetCaps();
            }
            //Mobile Platform
            return platform.CanUseAppium()
                //Mobile Platform
                ? platform.IsAnAppleDevice()
                    ? (new AppiumIOSCreator()).Create(platform, testName).GetCaps()
                    : (new AppiumAndroidCreator()).Create(platform, testName).GetCaps()
                //Devolve to WebDriver    
                : platform.IsAnAppleDevice()
                    ? (new WebDriverIOSCreator()).Create(platform, testName).GetCaps()
                    : (new WebDriverAndroidCreator()).Create(platform, testName).GetCaps();
        }
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 10th August 2014
 * 
 */